pkg/plugins/golang/v2/scaffolds/api.go

```diff
@@ -92,8 +79,15 @@ func (s *apiScaffolder) newUniverse() *model.Universe {
 }
 
 func (s *apiScaffolder) scaffold() error {
-	if s.doResource {
-		s.config.UpdateResources(s.resource.GVK())
+	// Check if we need to do the API and controller before updating with previously existing info
+	doAPI := s.resource.API != nil && s.resource.API.Version != ""
+	doController := s.resource.Controller
+
+	if doAPI {
+		// Update the known data about resource
+		if _, err := s.config.UpdateResources(s.resource); err != nil {
+			return fmt.Errorf("error updating resources in config: %w", err)
+		}
```

Comment:
```suggestion
	if _, err := s.config.UpdateResources(s.resource); err != nil {
		return fmt.Errorf("error updating resources in config: %w", err)
	}   
	

	if s.doAPI {
	
```

We need to `update the resource` always because we add the GKV in all scenarios. Then, it should be the first step.  
The `doAPI := s.resource.API != nil && s.resource.API.Version != ""` added here not make clear that we always update the resource for all scenarios and make very confuse know when it should scaffold an API. 

Then, `doController := s.resource.Controller` shows not required either. Why we are not using here the s.doController as before?

Reply:
`doAPI := s.resource.API != nil && s.resource.API.Version != ""` is the condition that has to be true in case we scaffold the resource:
As I already explained in https://github.com/kubernetes-sigs/kubebuilder/pull/1869#discussion_r539979264, `s.resource.API` can't be a nil pointer so we can't just use `doAPI := s.resource.API != nil`, we also have to check if the string is empty or not. Because if you go to `Options.NewResource`, you will see that this string is only set if `Options.DoAPI` is true.

`doAPI` and `doController` need to be considerd before updating:
Imagine that we first do a `kb create api ... --resource --controller=false`. Afterwards, we scaffold the controller with `kb create api ... --resource=false --controller`. One you update the resource, `s.resource.API != nil && s.resource.API.Version != ""` will be true because we already scaffolded this in the previous command, and therefore the second command call will think that he needs to scaffold the API again, resulting in errors because some files yield errors if they try to be scaffolded and they previously exist. This is the reason why we have to remember `doAPI` and `doController` before updating the resource.

Right now, v2 only updates the resource if you set the `--resource` flag, and thats exactly the behavior I left. Your request to move this out of the conditional would be the change that is not required and will mess with the v2 project.

Reply:
> As I already explained in #1869 (comment), s.resource.API can't be a nil pointer 
My point here is that IMO we should still using here the bool doAPI as it was before. The changes made here does not make the code better. Could we revert and keeps the original logic with the only changes that matter for us which is moving the `updateResources` to the top since now we are always updating the resources?  

Also, not that this logic should not change at all for v2. I mean, the update resources for v2 should still at the same line where it is currently in master. The purpose of this PR is to change the behaviour only for v3.